src/app/machines/views/MachineDetails/MachineHeader/MachineHeader.tsx

```diff
@@ -134,6 +141,18 @@ const MachineHeader = ({
                   const view = Object.values(MachineHeaderViews).find(
                     ([, actionName]) => actionName === action
                   );
+                  const [, actionFunction] =
+                    Object.entries(machineActions).find(
+                      ([key]) => key === action
+                    ) || [];
+                  if (
+                    (action === NodeActions.LOCK ||
+                      action === NodeActions.UNLOCK) &&
+                    actionFunction
+                  ) {
+                    dispatchForSelectedMachines(actionFunction);
+                    return;
+                  }
```

Comment:
IMO I feel like this might be cleaner, since we're not having to process through an object to get the action. Otherwise it's additional processing that we have to do every time the button is clicked

```suggestion
                  if (action === NodeActions.LOCK) {
                    dispatchForSelectedMachines(machineActions.lock);
                    return;
                  } else if (action === NodeActions.UNLOCK) {
                    dispatchForSelectedMachines(machineActions.unlock);
                    return;
                  }
```

Reply:
This makes sense